### DISCUSSION TITLE
Accessing environment variables in Svelte

### DISCUSSION THREAD

**Discussion by mattWalters0** [vercel/vercel#7016] [Help]
In development I am using dotenv to access environment variables from a .env filke like this in the rollup.config.js file:
```
plugins: [
replace({
  __api: JSON.stringify({
	  env: {
			  isProd: production,
			  ...config().parsed // attached the .env config
	  }
  }),
  delimiters: ['', '']
	}),
```

But in production, I want to refer to the environment variables configured in the Vercel UI:
<img width="666" alt="Screenshot 2021-11-16 at 12 39 15" src="https://user-images.githubusercontent.com/24442464/141987038-29eb3caa-1609-4a3c-bc78-33ba183545f2.png">

But I can't figure out how to access these??  Apologies if this is not clear.

**Comment by Rich-Harris** [comment]
If I understand correctly, `config().parsed` only includes the values in the `.env` file, whereas you need to inherit values from `process.env`. One alternative that should work is to use `dotenv/config` to add the contents of the `.env` file if it exists, so that `process.env` is correctly populated in dev or prod, then pass through the relevant values explicitly:

```js
import 'dotenv/config';

const {
  SVELTE_APP_SUPABASE_ANON_WHATEVER_THE_REST_OF_THIS_KEY_IS,
  SVELTE_APP_SUPABASE_URL
} = process.env;

export default {
  // ...
  plugins: [
    replace({
      __api: JSON.stringify({
        env: {
          isProd: production,
          SVELTE_APP_SUPABASE_ANON_WHATEVER_THE_REST_OF_THIS_KEY_IS,
          SVELTE_APP_SUPABASE_URL
        }
      })
    }),
    // ...
  ]
};
```
